features/storage/TESTS/kvstore/general_tests_phase_2/main.cpp

```diff
@@ -33,8 +33,8 @@
 using namespace utest::v1;
 using namespace mbed;
 
-#if !defined(TARGET_K64F) && !defined(TARGET_ARM_FM) && !defined(TARGET_MCU_PSOC6)
-#error [NOT_SUPPORTED] Kvstore API tests run only on K64F devices, Fastmodels, and PSoC 6
+#if !SECURESTORE_ENABLED
```

Comment:
Is this intended to be a new hard requirement? The existing test checks for and disables the securestore-specific tests when securestore is not enabled.

If this test is being made completely dependent on SecureStore, the other securestore related ifdefs (e.g. in kvstore_init()) should be removed.

Reply:
Yes. The purpose o this PR is to enable  all  storage tests if possible.
However if particular target doesn't  support all  necessary features test must be skipped and we can do nothing with this.

Reply:
The desire to enable all storage tests whenever possible doesn't seem to align with the check here.
Previously the test was structured so that it could run the FSStore and TDBStore tests on targets without SECURESTORE_ENABLED (if they were part of an otherwise limited list). Only the SecureStore dependent tests were disabled.
Now, the FSStore and TDBStore tests are also disabled when SecureStore is not enabled.

Are there not meaningful cases there SECURESTORE_ENABLED is false but TDBStore and FSStore do work?

Also, can you address this part of my previous comment?
>If this test is being made completely dependent on SecureStore, the other securestore related ifdefs (e.g. in kvstore_init()) should be removed.

Reply:
You right. I will skip only SecureStore. 
I need to fix TDBStore  because there are crashes on targets without TRNG